Source/Core/GoogleEarthEnterpriseMetadata.js

```diff
@@ -0,0 +1,579 @@
+        newInfo.ancestorHasTerrain = obj.ancestorHasTerrain;
+        newInfo.terrainState = obj.terrainState;
+
+        return newInfo;
+    };
+
+    /**
+     * Sets the parent for the tile
+     *
+     * @param {TileInformation} parent Parent tile
+     */
+    TileInformation.prototype.setParent = function(parent) {
+        this.ancestorHasTerrain = parent.ancestorHasTerrain || this.hasTerrain();
+    };
+
+    /**
+     * Gets whether a subtree is available
+     *
+     * @returns {Boolean} true if subtree is available, false otherwise.
+     */
+    TileInformation.prototype.hasSubtree = function() {
+        return isBitSet(this._bits, cacheFlagBitmask);
+    };
+
+    /**
+     * Gets whether imagery is available
+     *
+     * @returns {Boolean} true if imagery is available, false otherwise.
+     */
+    TileInformation.prototype.hasImagery = function() {
+        return isBitSet(this._bits, imageBitmask);
+    };
+
+    /**
+     * Gets whether terrain is available
+     *
+     * @returns {Boolean} true if terrain is available, false otherwise.
+     */
+    TileInformation.prototype.hasTerrain = function() {
+        return isBitSet(this._bits, terrainBitmask);
+    };
+
+    /**
+     * Gets whether any children are present
+     *
+     * @returns {Boolean} true if any children are available, false otherwise.
+     */
+    TileInformation.prototype.hasChildren = function() {
+        return isBitSet(this._bits, anyChildBitmask);
+    };
+
+    /**
+     * Gets whether a specified child is available
+     *
+     * @param {Number} index Index of child tile
+     *
+     * @returns {Boolean} true if child is available, false otherwise
+     */
+    TileInformation.prototype.hasChild = function(index) {
+        return isBitSet(this._bits, childrenBitmasks[index]);
+    };
+
+    /**
+     * Gets bitmask containing children
+     *
+     * @returns {Number} Children bitmask
+     */
+    TileInformation.prototype.getChildBitmask = function() {
+        return this._bits & anyChildBitmask;
+    };
+
+    GoogleEarthEnterpriseMetadata.TileInformation = TileInformation;
+
+    /**
+     * Provides metadata using the Google Earth Enterprise REST API. This is used by the
+     *
+     * @alias GoogleEarthEnterpriseMetadata
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {String} options.url The url of the Google Earth Enterprise server hosting the imagery.
+     * @param {Proxy} [options.proxy] A proxy to use for requests. This object is
+     *        expected to have a getURL function which returns the proxied URL, if needed.
+     *
+     * @see GoogleEarthEnterpriseImageryProvider
+     * @see GoogleEarthEnterpriseTerrainProvider
+     *
+     */
+    function GoogleEarthEnterpriseMetadata(options) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(options.url)) {
```

Comment:
We use `Check` for error checking now -> `Check.typeOf.string('options.url', options.url)`